### PR TITLE
feat: Add support for Filled Button

### LINKED ISSUE #264: feat: Add support for Filled Button
**Description**

Add support for Filled Button

**Requirements**

- [ ] Create a Mirai Filled Button Parser
- [ ] Add example to the Mirai gallery

### PR DESCRIPTION
**Pull request by iampranabray**
## Description

- Created a Mirai Filled Button Parser 
- Added example to the Mirai gallery 
- filled button json is added in assets file. different use case of filled button Screenshot is attached bellow .
Simulator Screenshot - iPhone 15 Pro - 2024-02-29 at 02 43 01

## Related Issues

Closes #264 

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

File: packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    MaterialStatesController? materialStatesController =
+        MaterialStatesController();
+
+    return FilledButton(
+      onPressed: model.onPressed == null ? null : () => model.onPressed,
+      onLongPress: model.onLongPress == null ? null : () => model.onLongPress,
+      onHover: (bool value) => value == false ? null : model.onHover,
+      onFocusChange: (bool value) =>
+          value == false ? null : model.onFocusChange,
+      style: model.style?.parseElevated(context),
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart:27]: We should not use the `parseElevated`. It is for the elevated button. Can we create one Filled Style parser?

File: packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      autofocus: model.autofocus,
+      clipBehavior: model.clipBehavior,
+      focusNode: focusNode,
+      statesController: materialStatesController,
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart:31]: We don't need `focusNode` and `statesController` as of now. We can remove it :)

**Review by divyanshub024 — CHANGES_REQUESTED** [review]
Hey @iampranabray 👋🏻 

Once again, Thank you for such amazing contributions. 

I have some suggestions, please check my comments.

File: packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart (hunk 1 above)
Comment (iampranabray) [reply packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart:27]: Ah mistaken. Sure I will replace ;)

File: packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart (hunk 2 above)
Comment (iampranabray) [reply packages/mirai/lib/src/parsers/mirai_filled_button/mirai_filled_button_parser.dart:31]: Sure I will remove .

**Comment by divyanshub024** [comment]
@iampranabray Can you please resolve these conflicts?

**Comment by iampranabray** [comment]
Yha sure

**Comment by iampranabray** [comment]
> @iampranabray Can you please resolve these conflicts? 

Yha sure

**Review by divyanshub024 — APPROVED** [review]
Thank you for this amazing contribution @iampranabray 🎉

**Pull request merged**